cpp/src/arrow/dataset/file_parquet.cc

```diff
@@ -538,17 +556,17 @@ Status ParquetFileFragment::EnsureCompleteMetadata(parquet::arrow::FileReader* r
   physical_schema_ = std::move(schema);
 
   std::shared_ptr<parquet::FileMetaData> metadata = reader->parquet_reader()->metadata();
-  int num_row_groups = metadata->num_row_groups();
```

Comment:
I think we still might need to keep this. Below, it was used for checking that the ids are not out of bounds. But using `num_row_groups_`, which can be smaller than the max number of row groups in the file, is the wrong number to check this, I think?